packages/block-library/src/navigation/view.js

```diff
@@ -73,60 +73,77 @@ store( {
 	selectors: {
 		core: {
 			navigation: {
-				roleAttribute: ( { context, selectors } ) =>
-					context.core.navigation.overlay &&
-					selectors.core.navigation.isMenuOpen( { context } )
+				roleAttribute: ( store ) => {
+					const { context, selectors } = store;
+					return context.core.navigation.type === 'overlay' &&
+						selectors.core.navigation.isMenuOpen( store )
 						? 'dialog'
-						: '',
+						: '';
+				},
 				isMenuOpen: ( { context } ) =>
-					// The menu is opened if either `click` or `hover` is true.
-					Object.values( context.core.navigation.isMenuOpen ).filter(
-						Boolean
-					).length > 0,
+					// The menu is opened if either `click`, `hover` or `focus` is true.
+					Object.values(
+						context.core.navigation[
+							context.core.navigation.type === 'overlay'
+								? 'overlayOpenedBy'
+								: 'submenuOpenedBy'
+						]
+					).filter( Boolean ).length > 0,
+				menuOpenBy: ( { context } ) =>
+					context.core.navigation[
+						context.core.navigation.type === 'overlay'
+							? 'overlayOpenedBy'
+							: 'submenuOpenedBy'
+					],
 			},
 		},
 	},
 	actions: {
 		core: {
 			navigation: {
-				openMenuOnHover( args ) {
-					openMenu( args, 'hover' );
+				openMenuOnHover( store ) {
+					const { navigation } = store.context.core;
+					if (
+						navigation.type === 'submenu' &&
+						// Only open on hover if the overlay is closed.
+						Object.values(
+							navigation.overlayOpenedBy || {}
+						).filter( Boolean ).length === 0
```

Comment:
This is the main change of this PR, where we check if the parent overlay is opened or not.

Reply:
Would it be safe to include a global variable to check if the overlay is opened? I can imagine other use cases where it could be handy to check `store.state.core.navigation.overlayMenuOpened`, or something similar.